src/firefly/js/templates/common/ttFeatureWatchers.js

```diff
@@ -190,37 +237,33 @@ export const PrepareDownload = React.memo(({table_id, tbl_title, viewerId, showF
         <>
             {loading && <ToolbarButton enabled={false} variant={'soft'} color='warning' text={downloadType === 'script' ? 'Generate Download Script' : 'Prepare Download'}/>}
             {!loading &&
-                <Stack>
-                    <DownloadButton key={keysToWatch}
-                        buttonText = {downloadType === 'script' ? 'Generate Download Script' : 'Prepare Download'}>
-                        <DownloadOptionPanel {...{
-                            updateSearchRequest,
-                            groupKey: tbl_id,
-                            tbl_id,
-                            downloadType,
-                            dlParams: {
-                                FileGroupProcessor:'ObsCorePackager',
-                                worker: downloadType === 'script' ? 'DownloadScriptWorker' : 'PackagingWorker',
-                                dlCutout: 'orig',
-                                position,
-                                cutoutValue,
-                                generateDownloadFileName,
-                                datalinkServiceDescriptor: isDatalinkSerDesc,
-                                viewerId,
-                                DataSource: dataSource,
-                                help_id:'table.obsCorePackage',
-                                Title: fileName ? fileName+ `_${baseFileName}` : `${baseFileName}`
-                            }}}>
-                            {!isDatalink && <Stack spacing={1}>
-                                <CheckboxGroupInputField
-                                    fieldKey='productTypes'
-                                        options={dynamicOptions}
-                                        initialState={{value: '*'}}
-                                        label='Products to Download: ' />
-                            </Stack>}
-                        </DownloadOptionPanel>
-                    </DownloadButton>
-                </Stack>
+                <FieldGroup groupKey={tbl_id}>
+                    <Stack>
+                        <DownloadButton key={keysToWatch}
+                            buttonText = {downloadType === 'script' ? 'Generate Download Script' : 'Prepare Download'}>
+                            <DownloadOptionPanel {...{
```

Comment:
Minor:  it's a good idea to move this into a separate component.  It makes it easier to read as well as identifying props(input).

Reply:
Not sure about this one, let's talk briefly on Monday morning before the meeting?